### PR TITLE
feat(tests): add `x/ecocredit` gRPC REST tests

### PR DESCRIPTION
**Pull request by aleem1314**
## Description

Closes: #541
Closes: #547 

 * [x] Update swagger-docs.
 * [x] Add gRPC-gateway integration tests.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [x] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [x] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [x] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [x] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [x] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [x] reviewed tests and test coverage
- [x] manually tested (if applicable)

### CODE REVIEW

File: x/ecocredit/client/testsuite/grpc.go [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+	testCases := []struct {
+		name       string
+		url        string
+		numBatches int
+		expErr     bool
+		errMsg     string
+	}{
+		{
+			"invalid class-id",
+			fmt.Sprintf("%s/regen/ecocredit/v1alpha1/batches?class_id=%s", val.APIAddress, "abcd"),
```
Comment (aleem1314) [inline x/ecocredit/client/testsuite/grpc.go:126]: Shall we update `credit-batches` endpoints? I think `/regen/ecocredit/v1alpha1/classes/{class_id}/batches` makes more sense.

Comment (ryanchristo) [reply x/ecocredit/client/testsuite/grpc.go:126]: Either way seems reasonable to me.

File: x/ecocredit/client/testsuite/grpc.go [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+			"denomination didn't match the format",
+		},
+		{
+			"invalid account address",
+			fmt.Sprintf("%s/regen/ecocredit/v1alpha1/batches/%s/balance/%s", val.APIAddress, "C01-20210101-20210201-001", "abcd"),
+			true,
+			"decoding bech32 failed",
+		},
+		{
+			"valid test",
```
Comment (ryanchristo) [inline x/ecocredit/client/testsuite/grpc.go:270]:
```suggestion
			"valid request",
```

File: x/ecocredit/client/testsuite/grpc.go [hunk 3]
```diff
@@ -0,0 +1,344 @@
...
+		errMsg string
+	}{
+		{
+			"invalid batch-denom",
+			fmt.Sprintf("%s/regen/ecocredit/v1alpha1/batches/%s/supply", val.APIAddress, "abcd"),
+			true,
+			"denomination didn't match the format",
+		},
+		{
+			"valid test",
```
Comment (ryanchristo) [inline x/ecocredit/client/testsuite/grpc.go:316]:
```suggestion
			"valid request",
```

File: x/ecocredit/client/testsuite/grpc.go [hunk 4]
```diff
@@ -0,0 +1,344 @@
...
+		},
+		{
+			"no batches found",
+			fmt.Sprintf("%s/regen/ecocredit/v1alpha1/batches/%s", val.APIAddress, "A00-00000000-00000000-000"),
+			true,
+			"not found",
+			"",
+		},
+		{
+			"valid test",
```
Comment (ryanchristo) [inline x/ecocredit/client/testsuite/grpc.go:201]:
```suggestion
			"valid request",
```

**Review by ryanchristo — APPROVED** [review]
tACK. Double checked gRPC routes. Tests look great. Nice work!

File: x/ecocredit/client/testsuite/grpc.go [hunk 5]
```diff
@@ -0,0 +1,344 @@
...
+		{
+			"valid request",
+			fmt.Sprintf("%s/regen/ecocredit/v1alpha1/classes/%s/batches", val.APIAddress, "C01"),
+			4,
+			false,
+			"",
+		},
+		{
+			"valid request with pagination",
+			fmt.Sprintf("%s/regen/ecocredit/v1alpha1/classes/%sbatches?pagination.limit=2", val.APIAddress, "C01"),
```
Comment (aleem1314) [inline x/ecocredit/client/testsuite/grpc.go:147]:
```suggestion
			fmt.Sprintf("%s/regen/ecocredit/v1alpha1/classes/%s/batches?pagination.limit=2", val.APIAddress, "C01"),
```

**Review by atheeshp — APPROVED** [review]

**Review by likhita-809 — APPROVED** [review]

**Pull request merged**